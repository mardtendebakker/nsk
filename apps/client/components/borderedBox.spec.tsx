import { render } from '@testing-library/react';
import { Box, BoxProps } from '@mui/material';
import BorderedBox from './borderedBox';

jest.mock('@mui/material/Box', () => jest.fn(() => null));

describe('BorderedBox', () => {
  const props: BoxProps = {
    sx: {
      border: jest.fn(),
      borderRadius: '0.5rem',
      bgcolor: 'red',
      legend: { display: 'none' },
      px: '.5rem',
    },
  };

  it('matches snapshot', () => {
    const { asFragment } = render(<BorderedBox {...props}><b>test</b></BorderedBox>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('passes props to BaseBorderedBox', () => {
    render(<BorderedBox {...props}><b>test</b></BorderedBox>);

    expect(Box).toBeCalledWith({ children: <b>test</b>, ...props }, {});
  });
});
