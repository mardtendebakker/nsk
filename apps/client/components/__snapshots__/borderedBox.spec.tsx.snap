// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderedBox matches snapshot 1`] = `<DocumentFragment />`;
